lib/init.js

```
@@ -0,0 +1,141 @@
+      loadDeps();
+    };
+    depCallbackID = window.requestIdleCallback(installLinterDeps);
+
+    this.idleCallbacks.add(depCallbackID);
+
+    const oldPath = atom.config.get('linter-luacheck.executable');
+    if (oldPath !== undefined) {
+      atom.config.unset('linter-luacheck.executable');
+      if (oldPath !== 'luacheck') {
+        // If the old config wasn't set to the default migrate it over
+        atom.config.set('linter-luacheck.executablePath', oldPath);
+      }
+    }
+
+    this.subscriptions = new CompositeDisposable();
+
+    this.subscriptions.add(
+      atom.config.observe('linter-luacheck.executablePath', (value) => {
+        this.executablePath = value;
+      }),
+      atom.config.observe('linter-luacheck.ignore', (value) => {
+        this.ignore = value;
+      }),
+      atom.config.observe('linter-luacheck.globals', (value) => {
+        this.globals = value;
+      }),
+      atom.config.observe('linter-luacheck.standards', (value) => {
+        this.standards = value;
+      })
+    );
+  },
+
+  deactivate() {
+    this.idleCallbacks.forEach(callbackID =>
+      window.cancelIdleCallback(callbackID)
+    );
+    this.idleCallbacks.clear();
+
+    this.subscriptions.dispose();
+  },
+
+  makeParameters(file) {
+    let params = [
+      '-', '--no-color', '--codes',
+      '--ranges', `--filename=${file}`
+    ];
+
+    params = checkedOption(params, '--globals', this.globals);
+    params = checkedOption(params, '--ignore', this.ignore);
+
+    if (this.standards.length > 0) {
+      params.push('--std');
+      params.push(this.standards.join('+'));
+    }
+
+    return params;
+  },
+
+  provideLinter() {
+    return {
+      name: 'Luacheck',
+      grammarScopes: ['source.lua'],
+      scope: 'file',
+      lintsOnChange: true,
+
+      lint: async (editor) => {
+        const file = editor.getPath();
+        const text = editor.getText();
+
+        if (!file || !text) {
```

Does `luacheck` crash if you try to run it on empty content? I guess I should have explained the other comment in more detail.

If `luacheck` _does_ crash on an empty file, it's probably better to return `[]` instead of `null`. Sorry for the confusion!